File: Backend/clinic/src/main/java/com/example/demo/entity/Doctor.java
package com.example.demo.entity;

import jakarta.persistence.*;
import lombok.Data;
import java.math.BigDecimal;

@Entity
@Table(name = "doctors")
@Data
public class Doctor {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "user_id", unique = true)
    private Long userId;

    @Column(name = "specialty_id")
    private Long specialtyId;

    @Column(name = "full_name", nullable = false)
    private String fullName;

    private String degree; 

    @Column(name = "room_number")
    private String roomNumber;

    @Column(name = "consultation_fee")
    private BigDecimal consultationFee;

    private String bio;

    @Column(name = "avatar_url")
    private String avatarUrl;

    private Double rating;

    @Transient
    private String specialtyName;
}

File: BenhNhan/myapp/lib/models/doctor.dart
class DoctorModel {
  final int id;
  final String fullName; 
  final int specialtyId;
  final String avatarUrl;
  final String bio;
  final String degree;
  final double rating;
  final String specialtyName;

  DoctorModel({required this.id, required this.fullName, required this.specialtyId, required this.avatarUrl, required this.bio, required this.degree, required this.rating, required this.specialtyName});

  factory DoctorModel.fromJson(Map<String, dynamic> json) {
    return DoctorModel(
      id: json['id'],
      fullName: json['fullName'],
      specialtyId: json['specialtyId'],
      avatarUrl: json['avatarUrl'] ?? '',
      bio: json['bio'] ?? '',
      degree: json['degree'] ?? '',
      rating: (json['rating'] ?? 0.0).toDouble(),
      specialtyName: json['specialtyName'] ?? '',
    );
  }
}